lib/core/test/unit/Cardano/Wallet/Types/Read/Tx/CBORSpec.hs

```diff
@@ -0,0 +1,31 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# OPTIONS_GHC -Wno-unused-top-binds #-}
+
+module Cardano.Wallet.Types.Read.Tx.CBORSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Api
+    ( CardanoEra (..) )
+import qualified Cardano.Chain.UTxO as Byron
+import Cardano.Wallet.Types.Read.Tx
+    ( Tx (..) )
+import Cardano.Wallet.Types.Read.Tx.CBOR
+    ( getTxCBOR, parseCBOR )
+import Test.Hspec
+    ( Spec, describe, it, parallel, pendingWith )
+import Test.QuickCheck
+    ( Property, (===) )
+
+spec :: Spec
+spec = describe "Cardano.Wallet.Types.Read.Tx.CBOR" $ do
+    parallel $ describe "TxCBOR encoding" $ do
+        it  "roundtrips properly" $ do
+            pendingWith "we need generators for the various era transactions"
```

Comment:
Instead of generating transactions randomly, I think it's sufficient to test on a couple of golden CBOR strings. After all, the `ToCBOR`, `FromCBOR` instances are written and tested by cardano-ledger — we're mostly just double checking that we picked the right functions.

I.e. I would suggest to test
```
(getTxCBOR <$> parseCBOR cbor) == Right cbor
```
for a couple of specific `cbor` values.

(Checking that the `cbor` is the same also helps with the issue that, in principle, there exist multiple CBOR encodings for the same `Tx` value, because e.g. lists can be encoded in two different ways.)

Reply:
I didn't find a cbor for byron. Other eras are added